connector/connect/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -62,3 +65,39 @@ message CreateScalarFunction {
     FUNCTION_LANGUAGE_SCALA = 3;
   }
 }
+
+// As writes are not directly handled during analysis and planning, they are modeled as commands.
+message WriteOperation {
+  // The output of the `input` relation will be persisted according to the options.
+  Relation input = 1;
+  // Format value according to the Spark documentation. Examples are: text, parquet, delta.
+  string format = 2;
+  // The destination of the write operation must be either a path or a table.
+  oneof save_type {
+    string path = 3;
+    string table_name = 4;
+  }
+  SaveMode mode = 5;
+  // List of columns to sort the output by.
+  repeated string sortColumnNames = 6;
+  // List of columns for partitioning.
+  repeated string partitionByColumns = 7;
```

Comment:
no biggie but I would match the name as what DataFrameWriter has. e.g.) `partitioningColumns`, `bucketColumnNames`, `sortColumnNames` and `extraOptions`.

Reply:
will do

Reply:
I changed this accordingly with a minor exception on the bucketing where I wrap this into the `BucketBy` message to have a better ux for setting the two values together.